### PR TITLE
Enable MCCAS by default in clang-cache

### PR DESCRIPTION
**Pull request by rastogishubham**
Enable MCCAS in clang-cache by default, disable it if environment variable DISABLE_MCCAS is set or a remote cache is used.

### CODE REVIEW

**Comment by rastogishubham** [comment]
@swift-ci please test

**Comment by rastogishubham** [comment]
@swift-ci please test llvm

File: clang/tools/driver/CacheLauncherMode.cpp [hunk 1]
```diff
@@ -105,6 +105,15 @@ static void addCommonArgs(bool ForDriver, SmallVectorImpl<const char *> &Args,
       }
     }
   };
+  auto RemotePath =
+      llvm::sys::Process::GetEnv("LLVM_CACHE_REMOTE_SERVICE_SOCKET_PATH");
+
+  if (!llvm::sys::Process::GetEnv("DISABLE_MCCAS") && !RemotePath) {
```
Comment (akyrtzi) [inline clang/tools/driver/CacheLauncherMode.cpp]: Use a prefixed environment variable name, like `CLANG_CACHE_DISABLE_MCCAS`

File: clang/tools/driver/CacheLauncherMode.cpp [hunk 2]
```diff
@@ -105,6 +105,15 @@ static void addCommonArgs(bool ForDriver, SmallVectorImpl<const char *> &Args,
...
+    addCC1Args({"-fcas-backend"});
```
Comment (akyrtzi) [inline clang/tools/driver/CacheLauncherMode.cpp]: These should go to `addLauncherArgs`, `addCommonArgs` is for flags common to launcher and dep-scan server.

File: clang/tools/driver/CacheLauncherMode.cpp [hunk 3]
```diff
@@ -105,6 +105,15 @@ static void addCommonArgs(bool ForDriver, SmallVectorImpl<const char *> &Args,
...
+    addCC1Args({"-fcas-backend-mode=verify"});
```
Comment (akyrtzi) [inline clang/tools/driver/CacheLauncherMode.cpp]: Why is this passed by default? Doesn't it introduce overhead that we don't want at production?

Comment (rastogishubham) [reply clang/tools/driver/CacheLauncherMode.cpp]: I just thought that we might want it enabled to catch any potential bugs in materialization

Comment (akyrtzi) [reply clang/tools/driver/CacheLauncherMode.cpp]: It shouldn't be always on if it has overhead, I'd suggest to add `CLANG_CACHE_VERIFY_MCCAS` env variable to control enablement.

Comment (rastogishubham) [reply clang/tools/driver/CacheLauncherMode.cpp]: sounds good!

File: clang/tools/driver/CacheLauncherMode.cpp [hunk 4]
```diff
@@ -125,6 +125,20 @@ static void addCommonArgs(bool ForDriver, SmallVectorImpl<const char *> &Args,
 /// Arguments specific to \p clang-cache compiler launcher functionality.
 static void addLauncherArgs(SmallVectorImpl<const char *> &Args,
                             llvm::StringSaver &Saver) {
+  auto RemotePath =
```
Comment (akyrtzi) [inline clang/tools/driver/CacheLauncherMode.cpp]: Use this variable again below where it again checks for `LLVM_CACHE_REMOTE_SERVICE_SOCKET_PATH`.

File: clang/tools/driver/CacheLauncherMode.cpp [hunk 5]
```diff
@@ -125,6 +125,20 @@ static void addCommonArgs(bool ForDriver, SmallVectorImpl<const char *> &Args,
...
+      llvm::sys::Process::GetEnv("LLVM_CACHE_REMOTE_SERVICE_SOCKET_PATH");
+
+  if (!llvm::sys::Process::GetEnv("CLANG_CACHE_DISABLE_MCCAS") && !RemotePath) {
```
Comment (akyrtzi) [inline clang/tools/driver/CacheLauncherMode.cpp]: It doesn't matter functionally, but aesthetically I'd prefer to add these options after the other existing options that get added in this function, not having them the first thing at the top of the function.

**Review by akyrtzi — APPROVED** [review]
Some minor nitpicks, otherwise LGTM.

**Pull request merged**